File: src/src/main/hackerrank/helper/java/PrintStream.java
package hackerrank.helper.java;

import java.util.LinkedList;
import java.util.List;
import java.util.Objects;

public final class PrintStream {

    private LinkedList<String> _output = new LinkedList<>();

    public void setOutputData(List<String> in) {
        in.forEach((x) -> _output.add(x));
    }

    public void println() {
    }

    public void println(Object any) {

        if (!_output.isEmpty()) {
            String elem = _output.pop();
            assert Objects.equals(any, elem) : any.toString() + " not equals " + elem;
        }
    }
}

File: src/src/main/scala/hackerrank/helper/PrintStream.scala
package hackerrank.helper

import scala.collection.mutable

class PrintStream {

  private val output: mutable.Queue[String] = mutable.Queue()

  def setOutputData(in: Seq[String]) = in.foreach(x => output.enqueue(x))
  def assertIsEmpty() = assert(output.isEmpty, "Output is not empty")
  def println() = {}
  def println(x: Any) = {
    if (output.nonEmpty) {
      val elem = output.dequeue
      assert(x.toString == elem, s"$x not equals $elem")
      Predef.println(s"$x equals $elem")
    }
  }
}
